Backend/src/API/REST/routes/project/projectRouter.ts

```diff
@@ -0,0 +1,46 @@
+import express, {Router} from "express";
+import { projectModel } from "../../../../database/models/project";
+import { userModel } from "../../../../database/models/user";
+import { visibilityStatus } from "../../../../database/models/project";
+import check_auth from "../../middleware/check_auth";
+
+export default function project_route() : Router {
+    let router = express.Router();
+
+    router
+        .post("/projects", check_auth(), async (req: any, res) => {
+            
+            const {title, description, visibility} = req.body;
+            const admin_user_id = req.user_id;
+
+            if (!(title && description && visibility && admin_user_id))
+                return res.status(400).send("Missing parameters");
+            
+            if(!(Object.values(visibilityStatus).includes(visibility)) )
+                return res.status(400).send("Visibility parameter is wrong format. You should type one of the followings: PRIVATE, PUBLIC, BY_INSTITUTION");
+
+            if (await projectModel.findOne({title}))
+                return res.status(409).send("Project with the given name already exists!");
```

Comment:
What if projects belong to different institutions, should this also hold true

Reply:
Good point, that is exactly the case. This need to be changed.